### PR TITLE
Bug 1856714: Adding checks for rescued tasks during reconciliation

### PR DESCRIPTION
**Pull request by VenkatRamaraju**
**Description of the change:**
Added a check to determine whether or not a task has been rescued.

**Motivation for the change:**
Avoid adding rescued tasks to the list of failures during reconciliation.
Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1856714
Backporting the PR https://github.com/operator-framework/operator-sdk/pull/3650

### CODE REVIEW

File: pkg/ansible/controller/reconcile.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 				return reconcile.Result{}, err
```
Comment (camilamacedo86) [inline pkg/ansible/controller/reconcile.go:186]: It is missing the fragment with the bug fix which will be used to gen the CHANGELOG.
if it was not added in master we also need a PR with it there as well.

Comment (VenkatRamaraju) [reply pkg/ansible/controller/reconcile.go:186]: Oh ok. I'll make that change here & open a PR on master as well then. Thanks.

File: changelog/fragments/rescue-reconciliation.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+entries:
+  - description: >
+      Added checks for rescued tasks during reconciliation
+      to avoid adding rescued tasks to the list of failures.
```
Comment (camilamacedo86) [inline changelog/fragments/rescue-reconciliation.yaml:4]:
```suggestion
      Stop to reconcile tasks in Ansible-based operators when the event raised is a rescue.
```

File: changelog/fragments/rescue-reconciliation.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+    kind: "addition"
+
+    breaking: false
```
Comment (camilamacedo86) [inline changelog/fragments/rescue-reconciliation.yaml:8]:
```suggestion
    breaking: bugfix
```

File: changelog/fragments/rescue-reconciliation.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+entries:
+  - description: >
+      Added checks for rescued tasks during reconciliation
+      to avoid adding rescued tasks to the list of failures.
+
+    kind: "addition"
```
Comment (camilamacedo86) [inline changelog/fragments/rescue-reconciliation.yaml:6]:
```suggestion
    kind: "bugfix"
```

Comment (camilamacedo86) [reply changelog/fragments/rescue-reconciliation.yaml:6]: Same changes to the master one.

File: internal/util/projutil/project_util.go [hunk 5]
```diff
@@ -315,7 +315,7 @@ func CheckGoModules() error {
 	}
 	if !goModOn {
 		return fmt.Errorf(`using go modules requires GO111MODULE="on", "auto", or unset.` +
-			` More info: https://sdk.operatorframework.io/docs/golang/quickstart/#a-note-on-dependency-management`)
+			` More info: https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/`)
```
Comment (camilamacedo86) [inline internal/util/projutil/project_util.go:318]:
```suggestion
			` More info: https://v0-19-x.sdk.operatorframework.io/docs/golang/quickstart/#a-note-on-dependency-management`)
```

Comment (camilamacedo86) [reply internal/util/projutil/project_util.go:318]: we need to point out for the doc of this branch 0.19. Also, this is the places where has the info ^

File: CHANGELOG.md [hunk 6]
```diff
@@ -20,7 +20,7 @@
 ### Additions
 
 - Add "panic" level for --zap-stacktrace-level (allows "debug", "info", "error", "panic"). ([#3040](https://github.com/operator-framework/operator-sdk/pull/3040))
-- The `operator-sdk` binary has a new CLI workflow and project layout for scaffolding Go operators that is aligned with Kubebuilder's CLI and project layout. See the new [Quickstart Guide](https://sdk.operatorframework.io/docs/golang/quickstart/) and the new [CLI reference](https://v0-19-x.sdk.operatorframework.io/docs/new-cli/) for more details. ([#3190](https://github.com/operator-framework/operator-sdk/pull/3190))
+- The `operator-sdk` binary has a new CLI workflow and project layout for scaffolding Go operators that is aligned with Kubebuilder's CLI and project layout. See the new [Quickstart Guide](https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/) and the new [CLI reference](https://v0-19-x.sdk.operatorframework.io/docs/new-cli/) for more details. ([#3190](https://github.com/operator-framework/operator-sdk/pull/3190))
```
Comment (camilamacedo86) [inline CHANGELOG.md:23]: Same : https://v0-19-x.sdk.operatorframework.io/docs/golang/quickstart/

File: cmd/operator-sdk/main.go [hunk 7]
```diff
@@ -46,7 +46,7 @@ func main() {
 		if operatorType == projutil.OperatorTypeGo {
 			depMsg := "Operator SDK has a new CLI and project layout that is aligned with Kubebuilder.\n" +
 				"See `operator-sdk init -h` and the following doc on how to scaffold a new project:\n" +
-				"https://sdk.operatorframework.io/docs/golang/quickstart/\n" +
+				"https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/\n" +
```
Comment (camilamacedo86) [inline cmd/operator-sdk/main.go:49]: same : https://v0-19-x.sdk.operatorframework.io/docs/golang/quickstart/

**Pull request merged**